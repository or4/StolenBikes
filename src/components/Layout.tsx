import React from 'react';

interface Props {}

export class Layout extends React.PureComponent<Props> {
    public render() {
        return <div>{this.props.children}</div>;
    }
}
