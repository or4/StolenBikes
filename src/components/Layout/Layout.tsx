import React from 'react';
import styled from 'styled-components';
import { media } from 'ui/media';

const Container = styled.div`
    margin: 0 auto;

    ${media.mobile`
		padding: 15px;
  	`}

	${media.tablet`
		padding: 15px;
	`}

    ${media.desktop`
		max-width: 940px;
	`}
`;

export const Layout: React.FC = props => <Container>{props.children}</Container>;
