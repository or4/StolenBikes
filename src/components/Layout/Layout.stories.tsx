import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { Layout } from './Layout';

const stories = storiesOf('Layout', module);
stories.addDecorator(withKnobs);

stories.add('Layout', () => {
    return (
        <Layout>
            <div>{getKnobs()}</div>
        </Layout>
    );
});

function getKnobs() {
    return text('Layout', 'Layout test)');
}
