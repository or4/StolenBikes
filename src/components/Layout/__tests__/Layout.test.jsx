import React from 'react';
import { mount } from 'enzyme';
import { Layout } from '../Layout';

describe('Layout', () => {
    it('should render correctly', () => {
        const layout = (
            <Layout>
                <div>Layout test</div>
            </Layout>
        );

        const mounted = mount(layout);
        expect(mounted).toMatchSnapshot();
        mounted.unmount();
    });
});
