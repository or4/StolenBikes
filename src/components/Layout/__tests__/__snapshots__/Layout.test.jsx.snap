// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<Layout>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "iOjrob",
            "rules": Array [
              "
    margin: 0 auto;

    ",
              "
        @media ",
              "(max-width: ",
              "479",
              "px)",
              " {
            ",
              "
		padding: 8px;
  	",
              "
        }
    ",
              "

	",
              "
        @media ",
              "(min-width: ",
              "480",
              "px) and (max-width: ",
              "991",
              "px)",
              " {
            ",
              "
		padding: 16px;
	",
              "
        }
    ",
              "

    ",
              "
        @media ",
              "(min-width: ",
              "992",
              "px)",
              " {
            ",
              "
		max-width: 940px;
	",
              "
        }
    ",
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa iOjrob"
      >
        <div>
          Layout test
        </div>
      </div>
    </StyledComponent>
  </styled.div>
</Layout>
`;
